--- timeslice.cpp.orig	2013-09-26 17:05:04.000000000 +0200
+++ timeslice.cpp	2013-09-26 17:07:35.000000000 +0200
@@ -41,18 +41,15 @@
 #include <windows.h>
 #endif
 
-#ifdef __APPLE__
-#include <AGL/gl.h>
-#else
-#include <GL/gl.h>
-#endif
-
 #include <stdio.h>
+#include <algorithm>
+
 #include "ofxsImageEffect.h"
 #include "ofxsMultiThread.h"
 
 #include "../include/ofxsProcessing.H"
 
+using std::min; using std::max;
 
 // Base class for the RGBA and the Alpha processor
 class TimeSliceBase : public OFX::ImageProcessor {
@@ -109,7 +106,7 @@
 };
 
 // template to do the RGBA processing
-template <class PIX, int nComponents, int max>
+template <class PIX, int nComponents, int maximum>
 class TimeCombiner : public TimeSliceBase {
 public :
   // ctor
